### PR TITLE
attestation: verifier: tdx: Allow equals in kernel param values

### PR DESCRIPTION
**Pull request by jodh-intel**
Allow kernel parameter values to contain equal signs, for example:

```
module_foo=bar=baz,wibble_setting=2
```

Fixes: #163.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Replaces https://github.com/confidential-containers/attestation-service/pull/164.

File: attestation-service/attestation-service/src/verifier/tdx/claims.rs [hunk 1]
```diff
@@ -275,4 +272,229 @@ mod tests {
 
         assert_json_eq!(expected, claims);
     }
+
+    #[test]
+    fn test_parse_kernel_parameters() {
```
Comment (Xynnn007) [inline attestation-service/attestation-service/src/verifier/tdx/claims.rs:277]: I like the detailed test cases very much, thanks for this! Could you use https://crates.io/crates/rstest crate to make it more rusty?

**Review by fitzthum — APPROVED** [review]
Nice. Thanks for adding so many test cases. See Ding's comment about `rstest`.

File: attestation-service/attestation-service/src/verifier/tdx/claims.rs (hunk 1 above)
Comment (jodh-intel) [reply attestation-service/attestation-service/src/verifier/tdx/claims.rs]: @Xynnn007 - Branch updated.

**Comment by mythi** [comment]
The fixes link looks wrong

**Comment by fitzthum** [comment]
> The fixes link looks wrong

True. Probably because this PR was originally opened against the AS repo. We don't actually require the fixes thing in this repo.

**Comment by fitzthum** [comment]
> I mentioned it because I'm not able to re-open to continue the work. Are you able to do that? 

Ah, yes. I can reopen. Sorry about that.

**Pull request merged**